pkg/providers/aws/provider_postgres.go

```diff
@@ -319,12 +313,25 @@ func (p *PostgresProvider) createRDSInstance(ctx context.Context, cr *v1alpha1.P
 	}
 	if mi != nil {
 		if _, err = rdsSvc.ModifyDBInstance(mi); err != nil {
+			// AWS handles invalid state with the ErrCodeInvalidDBInstanceStateFault
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == rds.ErrCodeInvalidDBInstanceStateFault {
```

Comment:
Just one comment on this, if there is an `ErrCodeInvalidDBInstanceStateFault` we would end up instantly reconciling, hitting this error. Each time calling `ModifyDBInstance`, building up excess write API calls. 

Going by the ticket we have a list of `safe` states, can we check on those before calling `ModifyDBInstance`, similarly to how we expose metrics with `rdsInstanceStatusIsHealthy`

Reply:
Unfortunately we don't have a list of safe states. We have a list of all states.

Reply:
Checking what the InvalidDBInstanceStateFault checks for I found https://github.com/aws/aws-sdk-go/blob/5d8a3d6c71e822edf1e5b7037989283ddd7b1c8e/models/apis/docdb/2014-10-31/docs-2.json#L823-L824 which is the same as our original check. closing this pr